lxd/storage/drivers/driver_powerflex_volumes.go

```diff
@@ -0,0 +1,719 @@
+package drivers
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/canonical/lxd/lxd/backup"
+	"github.com/canonical/lxd/lxd/migration"
+	"github.com/canonical/lxd/lxd/operations"
+	"github.com/canonical/lxd/lxd/revert"
+	"github.com/canonical/lxd/lxd/storage/filesystem"
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/lxd/shared/api"
+	"github.com/canonical/lxd/shared/instancewriter"
+	"github.com/canonical/lxd/shared/logger"
+	"github.com/canonical/lxd/shared/units"
+	"github.com/canonical/lxd/shared/validate"
+)
+
+// factorGiB divides a byte size value into Gibibytes.
+const factorGiB = 1024 * 1024 * 1024
+
+// CreateVolume creates an empty volume and can optionally fill it by executing the supplied filler function.
+func (d *powerflex) CreateVolume(vol Volume, filler *VolumeFiller, op *operations.Operation) error {
+	revert := revert.New()
+	defer revert.Fail()
+
+	// Get raw the volume size in GiB.
+	// PowerFlex accepts values without unit only.
+	sizeBytes, err := units.ParseByteSizeString(vol.ConfigSize())
+	if err != nil {
+		return err
+	}
+
+	sizeGiB := sizeBytes / factorGiB
+
+	id, err := d.powerFlex().createVolume(d.getVolumeName(vol, ""), strconv.FormatInt(sizeGiB, 10), d.getVolumeType(vol), d.config["powerflex.pool"])
+	if err != nil {
+		return err
+	}
+
+	revert.Add(func() { _ = d.powerFlex().deleteVolume(id, "ONLY_ME") })
+
+	volumeFilesystem := vol.ConfigBlockFilesystem()
+	if vol.contentType == ContentTypeFS {
+		devPath, cleanup, err := d.getMappedDevPath(vol, true)
+		if err != nil {
+			return err
+		}
+
+		revert.Add(cleanup)
+
+		_, err = makeFSType(devPath, volumeFilesystem, nil)
+		if err != nil {
+			return err
+		}
+	}
+
+	// For VMs, also create the filesystem volume.
+	if vol.IsVMBlock() {
+		fsVol := vol.NewVMBlockFilesystemVolume()
```

Comment:
I think we should chat with Dell about this as to whether there is a more efficient way of storing the VM filesystem config drive as allocating 8GB for this seems wasteful.

Reply:
I have added it to the specification as a place to discuss it further.